[pradyuprasad] ip

pradyuprasad opened the pull request:
# 🌟 Introducing Orion: Your Personal Task Manager

Orion is a command-line-based task management application designed to help you stay organized and productive. With Orion, you can manage various tasks such as todos, deadlines, and events, all from the comfort of your terminal.

## Features

- **Task Management**:
  - Create tasks with specific descriptions and track them easily.
  - Organize your tasks into three categories: `TODO`, `DEADLINE`, and `EVENT`.

- **File Storage**:
  - Persist your tasks between sessions using a simple CSV-based storage system.
  - Automatically loads your tasks when you start the application.

- **Command Interface**:
  - Use natural language commands like `todo`, `deadline`, and `event` to interact with the app.
  - Mark tasks as completed or delete them when they are no longer needed.

## Usage Example

Here's how you can add a new deadline task in Orion:

```plaintext
// Command to add a deadline task
deadline Submit report /by 15/09/2024 2359
```

## Advanced Parsing Feature: Event Command Validation

Orion's Parser class includes advanced features for validating complex commands, such as adding an event with specific start and end times. Below is an example of how the `validateEventCommand` method ensures that the event command is correctly formatted:

```java
public class Example {
    public static void main(String[] args) {
        Parser parser = new Parser();

        // Example command input
        String[] commandParts = {"event", "Team Meeting /from 20/09/2024 1400 /to 20/09/2024 1600"};

        try {
            // Validate the event command
            EventDetails eventDetails = parser.validateEventCommand(commandParts);
            System.out.println("Valid command. Event: " + eventDetails.getDescription() +
                               " from " + eventDetails.getFrom() + " to " + eventDetails.getTo());
        } catch (InvalidEventException e) {
            System.err.println("Invalid event command: " + e.getMessage());
        }
    }
}
```

## What's Next?

1. **Enhance User Interface**:
   - Improve command feedback with more detailed messages.
   - Add support for more complex date formats.

2. **Expand Functionality**:
   - Implement recurring tasks to automate repetitive scheduling.
   - Introduce task prioritization to help users focus on the most important tasks.

3. **Increase Test Coverage**:
   - Ensure all edge cases are handled with appropriate unit tests.
   - Achieve 100% code coverage across all classes.

## Development Status

- [x] **Implement basic task management features**
- [x] **Add CSV storage support**
- [ ] *Implement recurring tasks*
- [ ] *Improve user interface and feedback*

## Additional Information

> "Orion is more than just a task manager; it's your daily productivity companion." – The Orion Team

For more information, visit the [Orion Github page](https://github.com/pradyuprasad/ip).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

pradyuprasad committed:
rename main file

pradyuprasad committed:
Completed Level 0. Rename, Greet, Exit

pradyuprasad committed:
Completed Level 1. Echo

pradyuprasad committed:
Completed Level 2. Add, List

pradyuprasad committed:
Completed Level 3. Mark as Done

pradyuprasad committed:
Completed Level 4. ToDos, Events, Deadlines

pradyuprasad committed:
Fixed bugs in Level 4. ToDos, Events, Deadlines

pradyuprasad committed:
Wrote test

pradyuprasad committed:
Level 5. Handle Errors

pradyuprasad committed:
A-Enums

pradyuprasad committed:
major refactor

pradyuprasad committed:
Refactor TaskManager for file-based persistence and error handling

- Added file-based persistence to TaskManager using tasks.csv
- Implemented methods to load and save tasks from/to CSV file
- Removed taskIdCounter and dynamically generated task IDs based on existing tasks
- Enhanced error handling with FileInitializationException for file operations
- Updated Orion class to initialize TaskManager with exception handling
- Refactored add, delete, markAsDone, and unmarkAsDone methods to ensure persistence
- Improved consistency by loading tasks from file before every operation

pradyuprasad committed:
Merge branch 'branch-Level-7' into master

- Implemented file-based persistence for tasks using CSV in TaskManager
- Added methods to load and save tasks from/to the CSV file
- Introduced FileInitializationException for robust error handling during file operations
- Refactored task management methods (add, delete, markAsDone, unmarkAsDone) to ensure persistence
- Updated Orion class to handle TaskManager initialization with file-based persistence

pradyuprasad committed:
Fix method name and add missing isValidIndex implementation

- Implemented isValidIndex method in TaskManager to correctly check task index validity
- Updated Validator class to use isValidIndex instead of the incorrect method name

pradyuprasad committed:
Add InvalidDateFormatException to handle invalid date format errors

pradyuprasad committed:
Add parseDateTime helper function to handle date and time parsing

pradyuprasad committed:
Update validateEventCommand to use parseDateTime for date and time parsing

pradyuprasad committed:
Update validateDeadlineCommand to support date and time parsing with optional time

pradyuprasad committed:
Refactor to use LocalDateTime for Deadlines and Events

- Removed `data/tasks.csv` to start fresh with correct date formats.
- Updated `Deadline` and `Event` classes to use `LocalDateTime` for date and time fields.
- Created `DeadlineDetails` and `EventDetails` classes to encapsulate and transfer task details.
- Refactored `Validator` to return `DeadlineDetails` and `EventDetails` objects.
- Adjusted `TaskManager` methods to work with the new details classes.
- Updated input.txt to use correct date formats for testing.

pradyuprasad committed:
Refactor TaskManager to use ISO-8601 date format and improve error handling for mark/unmark commands

pradyuprasad committed:
Merge branch 'branch-Level-8'

This merge incorporates the `branch-Level-8` enhancements into the `master` branch, bringing significant improvements and bug fixes to the codebase.

Key Enhancements:
1. **Date and Time Handling:**
   - Implemented support for ISO-8601 date and time format (`YYYY-MM-DDTHH:MM`) in the `TaskManager` class.
   - Ensured that all deadlines and event dates are stored and parsed using this standardized format, improving consistency and reducing potential parsing errors.
   - Updated the `Validator` class to validate user inputs correctly according to the new date format.

2. **Improved Error Handling:**
   - Enhanced the `InvalidMarkException` to provide clearer error messages, guiding users on the correct usage of the `mark` and `unmark` commands.
   - Refined the logic in the `Validator` class for handling `mark` and `unmark` commands, ensuring that invalid commands are caught and reported accurately.

3. **User Interface Updates:**
   - Adjusted the output formatting for task lists to display dates and times in a user-friendly manner while maintaining internal consistency with the ISO-8601 standard.
   - Ensured that tasks are correctly marked, unmarked, and deleted, with corresponding updates in the task list.

Additional Notes:
- This merge is crucial for aligning the `master` branch with the latest improvements in date and time handling, as well as providing a more robust and user-friendly interface.
- All relevant tests have been updated and verified to ensure that the new functionality works as expected across different scenarios.

pradyuprasad committed:
Rename TaskManager to TaskList

This commit refactors the task management system to align with
the A-MoreOOP requirements. Specifically:

- Renamed TaskManager class to TaskList
- Updated all references to TaskManager across the project
- Adjusted package structure from taskManager to taskList

This change is part of the broader refactoring effort to make
the code more object-oriented and modular. It's a step towards
implementing the required class structure for the A-MoreOOP task.

Future commits will address the creation of Ui and Storage classes,
as well as further refactoring of the main Orion (soon to be Duke) class.

pradyuprasad committed:
Rename Validator to Parser

- Changed Validator class to Parser
- Updated all references in Orion.java
- Moved Parser.java to correct package

Part of ongoing refactoring for A-MoreOOP task.

pradyuprasad committed:
Add UI class for user interaction handling

- Create new UI class in src/main/java/ui/UI.java
- Implement methods for displaying messages and reading user input
- Prepare for refactoring user interaction logic from Orion class

This change is part of the A-MoreOOP refactoring to improve
separation of concerns in the application.

pradyuprasad committed:
Implement Storage class and refactor TaskList

- Add new Storage class for file I/O operations
- Modify TaskList to use Storage for reading and writing tasks
- Update Orion to initialize both Storage and TaskList
- Propagate FileInitializationException in TaskList methods

This refactoring improves separation of concerns and modularity
in the task management system, aligning with A-MoreOOP principles.

pradyuprasad committed:
Merge branch 'branch-A-MoreOOP' into master

This merge implements the A-MoreOOP refactoring, significantly
improving the project's structure and adherence to OOP principles:

1. User Interface (UI):
   - Add new UI class to handle all user interactions
   - Implement methods for displaying messages and reading user input
   - Move UI-related code from Orion to the new UI class

2. Storage:
   - Implement new Storage class for file I/O operations
   - Add methods for reading from and writing to the data file
   - Encapsulate file initialization logic

3. Task Management:
   - Rename TaskManager to TaskList for clarity
   - Refactor TaskList to use the new Storage class
   - Update task manipulation methods to work with Storage

4. Input Parsing:
   - Rename Validator to Parser for more accurate representation
   - Adjust Parser methods to work with the refactored TaskList

5. Main Application (Orion):
   - Update to use new UI, Storage, TaskList, and Parser classes
   - Improve error handling with FileInitializationException

6. Exception Handling:
   - Propagate FileInitializationException in TaskList methods
   - Update exception handling in Orion to use new UI methods

These changes significantly improve the separation of concerns,
making the codebase more modular, maintainable, and extensible.
The refactored structure aligns with SOLID principles and sets
a foundation for future enhancements.

pradyuprasad committed:
Refactor project to organize all classes under the `orion` package

- Renamed the main package to `orion` and adjusted all subpackages accordingly.
- Updated all import statements to reflect the new package structure.
- Removed `data/tasks.csv` to clean up the data directory.
- Modified the `runtest.sh` script to accommodate the new package structure and ensure proper file compilation.
- Improved the organization and maintainability of the codebase by following a consistent package naming convention.

pradyuprasad committed:
Merge 'branch-A-Packages' into 'main'

Duke A-Packages: Organize classes into appropriate packages

This merge completes the A-Packages level task by organizing all Java
classes under the `orion` package, following the guidelines provided
in the SWE class.

- Organized all classes into the `orion` package for better structure.
- Created subpackages: `orion.commands`, `orion.parser`, `orion.task`,
  `orion.ui`, etc., to logically group related classes.
- Updated all import statements to match the new package structure.
- Removed `data/tasks.csv` to keep the data directory clean.
- Modified `runtest.sh` to accommodate the new structure, ensuring
  proper compilation and testing.

This refactor enhances code maintainability, aligning with the stretch
goal of dividing classes into multiple packages as the codebase grows.

pradyuprasad committed:
Merge branch 'add-gradle-support'

pradyuprasad committed:
Update main class and shadowJar configuration

- Set mainClass to `orion.Orion` in build.gradle to align with the correct entry point.
- Change shadowJar archive base name from `duke` to `orion`.
- Add `data/` directory to .gitignore to exclude runtime data from version control.
- Add the correct package declaration `package orion;` in Orion.java.

pradyuprasad committed:
Merge branch 'fix-issues' into master

This merge addresses various project configuration and structural issues:
- Corrected the main class reference in build.gradle to `orion.Orion`.
- Updated shadowJar configuration to use `orion` as the archive base name.
- Added the `data/` directory to .gitignore to prevent unnecessary files from being tracked.
- Ensured `Orion.java` has the appropriate package declaration.

These changes resolve the issues in the 'fix-issues' branch and align the project with its intended structure.

pradyuprasad committed:
Add JUnit tests for Parser and Deadline classes

- Added unit tests for the Parser class, covering validation of commands such as list, mark, unmark, todo, event, deadline, and delete.
- Added unit tests for the Deadline class, verifying correct construction, type icon retrieval, and string representation.
- Ensured that all tests handle relevant exceptions appropriately and validate expected outcomes.

These tests improve code coverage and ensure that core functionality behaves as expected.

pradyuprasad committed:
Merge branch 'branch-A-JUnit' into master

This merge adds comprehensive JUnit tests for the Parser and Deadline classes:
- ParserTest: Includes tests for command validation methods, such as list, mark, unmark, todo, event, deadline, and delete commands. These tests ensure that the parser correctly handles both valid and invalid input scenarios and raises appropriate exceptions.
- DeadlineTest: Verifies the proper construction of Deadline objects, checks the correctness of the type icon, and validates the string representation format.

These additions improve test coverage, ensuring robustness and reliability of core functionalities.

pradyuprasad committed:
Change build.gradle to allow JAR file generation

pradyuprasad committed:
Merge branch 'branch-A-Jar'

pradyuprasad committed:
Updated gitignore to ignore fish shell scripts

- Added *.fish to .gitignore to exclude fish shell scripts from being tracked.
- Fish shell scripts are used in development and not meant for
  production use

pradyuprasad committed:
Fix deadline task output formatting inconsistency in text UI tests

- Updated the expected output in text-ui-test/EXPECTED.TXT to correct the formatting for deadline tasks.
- Ensured consistent indentation and formatting across all task types.
- This change addresses inconsistencies between how deadline tasks were previously displayed versus other tasks.

pradyuprasad committed:
Refactor Orion and UI for streamlined formatting

- Removed `executeWithFormatting` calls from the Orion class and moved them into the respective methods in the UI class.
- Simplified the Orion class by delegating all output formatting responsibilities to the UI class.
- Updated UI methods (e.g., `showTaskAdded`, `showTaskList`, `showTaskMarked`) to include consistent formatting with `executeWithFormatting`.
- This refactoring enhances code readability and maintains a clear separation of concerns between the Orion class (logic) and the UI class (presentation).

pradyuprasad committed:
Merge branch 'fix-UI-errors' into master

- Refactored the UI class to centralize formatting and output handling.
- Removed redundant `executeWithFormatting` calls in the Orion class, simplifying the codebase.
- Ensured consistent task formatting across all command outputs.
- Improved code readability and maintainability by delegating UI responsibilities to the UI class.

pradyuprasad committed:
Add *.txt to .gitignore

- Exclude .txt files from version control as they are used during development, not in production.
- These files typically contain temporary data or commands intended for AI assistants and are not necessary to track in the repository.

pradyuprasad committed:
Add JavaDoc comments to Orion class methods

- Added detailed JavaDoc comments to the Orion class and its methods to improve code documentation.
- Provided descriptions for the main class, its entry point, and all command-handling methods.
- Documented the purpose, behavior, and parameters for each method to enhance code readability and maintainability.

pradyuprasad committed:
fix coding style violations by removing unecessary imports

pradyuprasad committed:
Add JavaDoc comments to Command enum and fromString method

The Command enum lacked documentation, which could lead to confusion
for future developers.

JavaDoc comments are added to describe the purpose of the Command enum
and its values. The fromString method now includes comments explaining
its functionality and how it handles invalid input.

pradyuprasad committed:
Refactor imports and format code in  classes and

- Organized imports across multiple classes to improve readability and maintain consistency.
- Applied consistent code formatting, including spacing and line breaks, in Parser, Storage, TaskList classes, and related test files.
- Ensured proper indentation and alignment of method signatures and blocks for better code structure.
- No functional changes were made; these updates focus on enhancing code clarity and maintainability.

pradyuprasad committed:
Add comments and improve exception messages

- Enhanced documentation for custom exceptions to provide clear explanations and usage instructions.
- Updated exception messages to include specific formatting details and usage examples.
- Ensured consistency in error messaging across different exception classes.
- Fixed formatting issues in exception messages for better readability.

pradyuprasad committed:
Add detailed documentation and enhance command validation in Parser

- Added detailed Javadoc comments for all command validation methods.
- Improved exception messages to provide clearer feedback on command errors.
- Added validation for todo, event, deadline, delete, mark, and unmark commands.
- Enhanced date and time parsing with improved error handling and validation.
- Updated method documentation to describe parameters, return values, and exceptions.

pradyuprasad committed:
Add Javadoc comments to Storage class

- Added Javadoc comments to the Storage class to provide detailed descriptions
  for methods and class-level documentation.
- Documented the purpose of the class, including its responsibilities for
  initializing, reading, and writing tasks to the storage file.

pradyuprasad committed:
Create JavaDoc for all classes in Task

pradyuprasad committed:
Add javadoc to TaskList

pradyuprasad committed:
make build and formatting changes

pradyuprasad committed:
update gitignore to remove txt files

pradyuprasad committed:
Add FIND command to enable task searching

Add a FIND command to the Command enum to allow Orion to parse
and handle user requests for finding tasks based on keywords in
their descriptions.

pradyuprasad committed:
Implement `FIND` command to search tasks

- Added `FIND` command to the `Command` enum to allow searching for tasks by keyword in the description.
- Updated `Orion` class with `handleFind` method to process the `FIND` command.
- Added `validateFindCommand` method in `Parser` to validate `FIND` command input.
- Extended `TaskList` class with `findTasks` method that filters tasks by a keyword using streams.
- Enhanced `UI` class to display matching tasks when the `find` parameter is set to true.
- Updated text UI tests to include test cases for the `FIND` command with various scenarios.

pradyuprasad committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDocs across various classes and methods to improve code documentation and maintainability.

- Detailed JavaDocs added to classes like `Orion`, `Command`, and `Storage`.
- Enhanced documentation for custom exceptions in the `orionExceptions` package.
- Updated `.gitignore` to exclude `.txt` files, preventing unnecessary files from being tracked.

These changes enhance code readability and help future developers understand the codebase more effectively.

pradyuprasad committed:
Merge branch 'branch-A-CodingStandard'

Improve code quality and documentation across Orion application:

- Add comprehensive Javadoc comments to all classes and methods
- Enhance code organization and readability
- Improve exception handling and error messages
- Refactor TaskList and Storage classes for better separation of concerns
- Update Parser class with more robust input validation
- Modify .gitignore to exclude .txt files
- Standardize import statements and code formatting

pradyuprasad committed:
Implement `FIND` command to search tasks

    - Added `FIND` command to the `Command` enum to allow searching for tasks by keyword in the description.
    - Updated `Orion` class with `handleFind` method to process the `FIND` command.
    - Added `validateFindCommand` method in `Parser` to validate `FIND` command input.
    - Extended `TaskList` class with `findTasks` method that filters tasks by a keyword using streams.
    - Enhanced `UI` class to display matching tasks when the `find` parameter is set to true.
    - Updated text UI tests to include test cases for the `FIND` command with various scenarios.

kaajinn commented on src/main/java/orion/storage/Storage.java:
```diff
@@ -0,0 +1,187 @@
+                boolean completed = Boolean.parseBoolean(parts[3]);
+
+                Task task = null;
+                switch (type) {
```
Should the indentation for the 'case' clauses be removed?

kaajinn commented on src/main/java/orion/Orion.java:
```diff
@@ -0,0 +1,208 @@
+            String[] parts = input.split(" ", 2);
+            Command command = Command.fromString(parts[0]);
+
+            switch (command) {
```
Should the indentation for the 'case' clauses be removed?

kaajinn commented on src/main/java/orion/task/Task.java:
```diff
@@ -0,0 +1,96 @@
+public abstract class Task {
+    private final int taskID;
+    private final String description;
+    private boolean completed;
```
Should this be changed to 'isComplete' or something similar to more closely adhere to the coding standard?

kaajinn commented on src/main/java/orion/commands/Command.java:
```diff
@@ -0,0 +1,36 @@
+ * recognize and execute.
+ * Each enum constant corresponds to a specific command that the user can input.
+ */
+public enum Command {
```
I like how clean this is 👍, very organized way of managing commands!

kaajinn reviewed (commented):
LGTM! Only very minor nits, otherwise a fantastic job!

kaajinn commented on src/main/java/orion/Orion.java:
```diff
@@ -0,0 +1,208 @@
+            String[] parts = input.split(" ", 2);
+            Command command = Command.fromString(parts[0]);
+
+            switch (command) {
+                case BYE:
+                    ui.showGoodbye();
+                    return;
+                case LIST:
+                    handleList(parts);
+                    break;
+                case MARK:
+                    handleMark(parts);
+                    break;
+                case UNMARK:
+                    handleUnmark(parts);
+                    break;
+                case TODO:
+                    handleTodo(parts);
+                    break;
+                case EVENT:
+                    handleEvent(parts);
+                    break;
+                case DEADLINE:
+                    handleDeadline(parts);
+                    break;
+                case DELETE:
+                    handleDelete(parts);
+                    break;
+                case FIND:
+                    handleFind(parts);
+                    break;
+                case UNKNOWN:
+                default:
+                    ui.showError("Unknown command: " + parts[0]);
+            }
```
Should the indentation for the 'case' clauses be removed?

kaajinn commented on src/main/java/orion/commands/Command.java:
```diff
@@ -0,0 +1,36 @@
+ * recognize and execute.
+ * Each enum constant corresponds to a specific command that the user can input.
+ */
+public enum Command {
+    LIST,
+    MARK,
+    UNMARK,
+    TODO,
+    DEADLINE,
+    EVENT,
+    DELETE,
+    BYE,
+    FIND,
+    UNKNOWN;
+
```
I like how clean this is 👍, very organized way of managing commands!

kaajinn commented on src/main/java/orion/storage/Storage.java:
```diff
@@ -0,0 +1,187 @@
+                boolean completed = Boolean.parseBoolean(parts[3]);
+
+                Task task = null;
+                switch (type) {
+                    case "TODO":
+                        task = new Todo(taskId, description);
+                        break;
+                    case "DEADLINE":
+                        LocalDateTime by = LocalDateTime.parse(parts[4]);
+                        task = new Deadline(taskId, description, by);
+                        break;
+                    case "EVENT":
+                        String[] eventTimes = parts[4].split("\\|");
+                        LocalDateTime from = LocalDateTime.parse(eventTimes[0]);
+                        LocalDateTime to = LocalDateTime.parse(eventTimes[1]);
+                        task = new Event(taskId, description, from, to);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unknown task type: " + type);
+                }
```
Should the indentation for the 'case' clauses be removed?

kaajinn reviewed (commented):
Fixed the formatting for some of the previous comments I left for easier reading. My bad!

clarissatjx commented:
Naming conventions in general were great, no major changes to be made (with regards to second PR review coding standards). Your code is really neat and easy to read, keep it up :)

pradyuprasad committed:
Add JavaFX setup and initial GUI components

pradyuprasad committed:
Fix errors in GUI

pradyuprasad committed:
fix image loading issues

pradyuprasad committed:
Implement GUI for Orion Task Manager

- Add JavaFX setup and initial GUI components
- Create MainWindow and DialogBox classes
- Implement basic task management functionality in GUI
- Fix image loading issues, convert WebP to PNG
- Update build.gradle for JavaFX support

This merge transforms Orion from a CLI to a GUI application,
providing a more user-friendly interface for task management."

pradyuprasad committed:
Update Orion and Parser classes to use varargs

- Refactored Orion.java to use varargs in methods where commands are passed.
- Updated Parser.java to accept TaskList and varargs in relevant methods (validateMarkAndUnMarkCommand, validateDeleteCommand).
- Modified ParserTest.java to properly test methods with varargs and TaskList.

pradyuprasad committed:
Merge branch 'branch-A-Varargs'

- Refactored Orion.java to use varargs in methods where commands are passed.
- Updated Parser.java to accept TaskList and varargs in relevant methods (validateMarkAndUnMarkCommand, validateDeleteCommand).
- Modified ParserTest.java to properly test methods with varargs and TaskList.

pradyuprasad committed:
Update build.gradle for clarity and configuration consistency

pradyuprasad committed:
Add assertions in TaskList, Storage

Add assertions in `TaskList` to ensure task list size is never negative
after loading, adding, or removing tasks. This helps in detecting
unexpected internal errors.

- Ensure tasks are not null before writing them to the file in `Storage`.
- Validate that tasks loaded from storage are not null.
- Ensure that the task is properly marked as completed when marking
  tasks in `markAsDone()`.
- Added an assertion to confirm that the task being deleted is not null
  in `deleteTask()`.

These assertions improve code robustness during development by ensuring
key internal state invariants.

pradyuprasad committed:
Simplify validation for mark/unmark command

The current implementation of the validateMarkAndUnMarkCommand method
contains redundant logic for command validation and task index
extraction.

- Reused validateCommandParts to validate the command type for "mark"
  and "unmark" to avoid duplication.
- Extracted task index validation into a separate method
  (extractTaskIndex) to simplify the logic and improve code reuse.
- Replaced inline task index validation with a dedicated
  validateTaskIndex method to check if the index is valid against
  TaskList.

These changes improve code readability and maintainability while
ensuring consistency across different commands.

pradyuprasad committed:
Remove unused UI class

The UI class in src/main/java/orion/ui/UI.java is not being used in the
current implementation of the project.

Unused classes in the codebase can lead to confusion and unnecessary
maintenance overhead. They can also make the codebase harder to
navigate and understand.

Let's remove this unused class to improve code cleanliness and reduce
potential confusion for other developers working on the project.

This removal:
* Decreases the overall complexity of the codebase
* Reduces the maintenance burden
* Improves code navigation by removing irrelevant files

pradyuprasad committed:
Add and configure Checkstyle plugin

Project currently lacks consistent code style enforcement.

Let's add Checkstyle plugin to improve code quality:
* Set version to 8.45
* Use custom rules file
* Enable HTML reports, disable XML reports

This will help maintain consistent coding standards.

pradyuprasad committed:
Reorder import statements in Storage class

Java imports are now ordered as follows:
* Standard Java imports
* Custom (orion) package imports
* Alphabetical order within each group

Reasons for change:
* Improve code readability
* Adhere to common Java conventions
* Facilitate easier code maintenance

pradyuprasad committed:
Reorganize imports for code clarity

- Moved import statements in Orion.java for better organization
- Removed unnecessary imports in TaskList.java

pradyuprasad committed:
Refactor package name for consistency

- Renamed package `taskList` to `tasklist` to ensure uniform naming conventions
- Updated all relevant import statements and package declarations

pradyuprasad committed:
Refactor switch statement

- Converted traditional switch statement to the enhanced switch expression with arrow syntax for cleaner code.

pradyuprasad committed:
Reorganize import statements in Parser.java

- Moved standard Java imports (java.time, java.util) above project-specific imports for better readability and adherence to coding conventions.

pradyuprasad committed:
Remove trailing spaces and standardize formatting

There were unnecessary trailing spaces and incorrect indentation in
the Storage.java and TaskList.java files, which violate the coding style
guidelines.

Trailing spaces were removed from various Javadoc comments. Additionally,
the indentation levels for `case` and `block` statements within switch
blocks were corrected to maintain consistency and readability.

These changes improve code quality by ensuring proper formatting, but no
functional changes were made. The fixes ensure better readability and
conformance to the style guidelines.

pradyuprasad committed:
fix minor error in import

pradyuprasad committed:
Change package name to lowercase

Renamed package `orion.orionExceptions` to `orion.orionexceptions`
to comply with naming conventions.

This change was necessary to align with Java’s package naming
requirements (all lowercase). Updated all relevant files, including
Orion.java, exception classes, and test files, to reflect this change.

The renaming ensures consistency and prevents potential issues
with case-sensitive systems and tools.

pradyuprasad committed:
Enhance Parser with Javadoc

- Added comprehensive JavaDoc comments to document the purpose and functionality of various methods in the Parser class.
- Improved clarity and maintainability by specifying method parameters, return values, and exceptions in Javadoc.
- Refactored `validateMarkAndUnMarkCommand` and `validateDeleteCommand` methods to accept a TaskList manager, allowing for better management of task indices.
- Introduced validation methods for new command types, including `find` and extended error handling for existing commands (`list`, `event`, `deadline`).
- Improved code readability and consistency by grouping related methods and adding descriptive documentation throughout the class.

These enhancements contribute to better code maintainability and help future developers understand the command validation process more effectively.

pradyuprasad committed:
Refactor variable names for consistency in Parser.java

Changed instances of 'split_parts' to 'splitParts' for improved readability
and consistency with Java naming conventions. This enhances the code's
clarity and maintains a uniform style across the Parser class.

pradyuprasad committed:
Improve code formatting in Storage.java

Reformatted the switch-case structure for task creation by aligning the cases
for better readability. This change enhances code clarity and maintainability
without affecting the functionality.

pradyuprasad committed:
Enhance Javadoc documentation in the Orion class

- Added comprehensive Javadoc comments to the Orion class methods
  to improve documentation and clarity.
- Descriptions for the class and key methods were provided,
  detailing their responsibilities, parameters, return values, and possible exceptions.
- Methods documented include:
  - Constructor
  - start method
  - getResponse method
  - handleList, handleMark, handleUnmark, handleTodo, handleEvent,
    handleDeadline, handleDelete, and handleFind methods.
- This documentation improvement will aid future developers in understanding
  the structure and functionality of the Orion application.

pradyuprasad committed:
Clean up JavaDoc comments in task classes

- Removed unnecessary blank lines in JavaDoc comments for the Deadline, Event, and Todo classes.
- Enhanced the readability of JavaDoc by maintaining consistent formatting.

These changes improve the overall clarity without altering the functionality of the code.

pradyuprasad committed:
Add documentation for Parser and UI components

- Introduced Javadoc comments for the `validateTodoCommand` method in the Parser class to explain its functionality and the expected command format.
- Added class-level Javadoc comments for `DialogBox` and `MainWindow` classes to describe their purpose and functionality within the application.
- Enhanced code readability and maintainability by providing clear documentation for critical components.

pradyuprasad committed:
Refactor Parser class: rearrange methods

- Moved `validateMarkAndUnMarkCommand` and `validateDeleteCommand` methods for better organization and readability.
- Updated method comments to enhance clarity.

pradyuprasad committed:
Enhance Documentation and Formatting

- Added JavaDoc comments to the Launcher class to describe its purpose and functionality.
- Improved error message formatting in InvalidMarkException to enhance readability.
- Added JavaDoc comments to the findTasks method in TaskList to specify parameters and exceptions.
- Minor formatting adjustments in DialogBox documentation for consistency.

These updates improve code clarity and provide better context for future developers.

pradyuprasad committed:
Merge pull request #1 from pradyuprasad/branch-A-CodeQuality

Branch a code quality

pradyuprasad committed:
Merge pull request #2 from pradyuprasad/branch-A-Assertions

Add assertions in TaskList, Storage

pradyuprasad committed:
Implement fuzzy search using Levenshtein distance

Task search functionality only matched exact keywords. This limitation
made it difficult for users to find tasks due to minor typos or
variations in the input.

Let's introduce fuzzy matching based on Levenshtein distance, allowing
tasks to be found even if the keyword provided is not an exact match.

The new method `levenshteinDistance` computes the distance between
task descriptions and the user-provided keyword, enabling flexible
search capabilities. This improves user experience by making the
search process more forgiving to input errors.

Task filtering logic is also streamlined to incorporate this fuzzy
matching approach.

pradyuprasad committed:
Merge pull request #3 from pradyuprasad/branch-C-BetterSearch

Implement fuzzy search using Levenshtein distance

pradyuprasad committed:
Ensure proper text wrapping and resizing in UI

- Add wrapText and setMaxWidth to the dialog Label in DialogBox.java
  to prevent text from being cut off.
- Modify DialogBox.fxml to use HBox.hgrow to allow the Label to expand
  and prevent truncation of long text.
- Adjust MainWindow.fxml to ensure the components resize properly
  when the window is resized, including setting anchors for the user
  input field and send button.

This change addresses the issue where long text strings in the dialog
box were truncated with ellipses (..), and ensures the UI adapts
to resizing for a better user experience.

pradyuprasad committed:
Merge branch 'branch-A-BetterGui'

Ensure proper text wrapping and resizing in UI

- Add wrapText and setMaxWidth to the dialog Label in DialogBox.java
  to prevent text from being cut off.
- Modify DialogBox.fxml to use HBox.hgrow to allow the Label to expand
  and prevent truncation of long text.
- Adjust MainWindow.fxml to ensure the components resize properly
  when the window is resized, including setting anchors for the user
  input field and send button.

This change addresses the issue where long text strings in the dialog
box were truncated with ellipses (..), and ensures the UI adapts
to resizing for a better user experience.

pradyuprasad committed:
remove .swp files from tracking

The .swp files generated by the editor are meant to be ignored as they
are temporary files.

Previously, the .Orion.java.swp file was tracked by Git. This commit
removes it from tracking while keeping it locally to avoid cluttering
the repository with temporary files.

Let's ensure that .swp files will not be tracked again by adding them
to the .gitignore file. This improves the repository's cleanliness and
reduces unnecessary changes being tracked.

pradyuprasad committed:
Refactor Orion Application Structure and UI Compon

- Updated the main class of the application from
'Launcher' to 'Orion'.
- Introduced a new DialogBox class to manage user
and Orion dialogues in a more structured way.
- Removed the previously used FXML components and
layouts, opting for direct JavaFX components in
the Main.java for better control and flexibility.
- Enhanced user interface setup, including image
loading for user and Orion.
- Removed unneeded imports and FXML resources
associated with the previous structure.
- Improved the handling of user inputs and
responses through a simplified main window setup.

This refactor aims to streamline the application
structure and improve the user experience as
topics and commands are processed by the UI.

pradyuprasad committed:
Add orion commands to chatbot

This commit modifies the GUI to add orion's commands.

Previously, we didn't have the orion commands in the GUI. This was a
major defect

This commit uses JavaFX to take the text from the Orion class and
display it in the chatbot GUI

pradyuprasad committed:
Reversed positions of User and Orion in GUI

Previous to this, both the User and Orion would come on the same side.

This commit changes orion GUI to have the chatbot come on the left
and the user come on the right in the GUI making it more like a
regular chatbox as expected by the prpjecty requirements

pradyuprasad committed:
Refactor GUI to use Java FXML

The previous implementation handled a lot of implementation
details in the Java code. This made the code cluttered and
hard to maintain.

This commit fixes that by rewriting the files pertaining to the
GUI code to be controllers and abstract all visual code to the
FXML files

pradyuprasad committed:
Merge branch 'fix-GUI-errors'

pradyuprasad committed:
Fix indentation errors in Orion.java

pradyuprasad committed:
Update README

pradyuprasad committed:
Fix error in README

pradyuprasad committed:
Merge branch 'update-README'

pradyuprasad committed:
Add title to GUI application

pradyuprasad committed:
Add UI.png to docs

pradyuprasad committed:
Write User Guide

pradyuprasad committed:
Renamed UI.png to Ui.png

pradyuprasad committed:
Removed temp.png after renaming to Ui.png

pradyuprasad committed:
Rename UI.png to Ui.png

pradyuprasad committed:
Update README.md with correct link

pradyuprasad committed:
Add welcome message with command list

- Implement getWelcomeMessage() in Orion class
- Display welcome message on app start
- Include command list with examples
- Update MainWindow to show welcome message

pradyuprasad committed:
Merge branch 'add-startup-message'

pradyuprasad committed:
Add Javadoc to GUI components

pradyuprasad committed:
Fix build.gradle for all OS
